test/testdata/infer/attached_class.rb

```diff
@@ -0,0 +1,26 @@
+# typed: true
+
+class Model
+  extend T::Sig
+
+  sig {params(args: T.untyped).returns(AttachedClass)}
+  def self.load(args)
+    T.unsafe(nil)
+  end
+end
+
+class A < Model
+end
+
+T.reveal_type(A.load(:foo)) # error: Revealed type: `A`
+
+T.reveal_type(T::Array[Integer].new) # error: Revealed type: `T::Array[Integer]`
+
+# Ensure that untyped generics still work correctly
+T.reveal_type(Array.new) # error: Revealed type: `T::Array[T.untyped]`
```

Comment:
Curious why you think these two tests are relevant? Don't you think these would have shown up as breakages somewhere else?

Reply:
They do show up elsewhere, in particular `test/testdata/resolver/untyped_generics.rb`. I thought they were nice to include here as well, as if they break in the future their colocation with other AttachedClass related tests might be helpful for directing debugging.

Reply:
gotcha, i'm fine keeping them.